sql/index.go

```diff
@@ -210,11 +210,16 @@ func (r *IndexRegistry) IndexByExpression(db string, expr Expression) Index {
 	defer r.mut.RUnlock()
 
 	for _, idx := range r.indexes {
-		if reflect.DeepEqual(idx.Expression(), expr) && idx.Database() == db {
-			r.retainIndex(db, idx.ID())
-			return idx
+		if idx.Database() == db {
+			for _, e := range idx.Expressions() {
```

Comment:
All the expressions must match, not only one. So maybe we need to change ` IndexByExpression(db string, expr Expression) Index` to ` IndexByExpression(db string, expr ...Expression) Index` and use `exprListsEqual`